dataprofiler/profilers/unstructured_text_profile.py

```diff
@@ -0,0 +1,223 @@
+        }
+
+        self.__calculations = {
+            "vocab": TextProfiler._update_vocab,
+            "words": TextProfiler._update_words,
+        }
+        BaseColumnProfiler._filter_properties_w_options(self.__calculations, options)
+
+    def _add_helper(self, other1, other2):
+        """
+        Merges the properties of two BaseColumnProfile objects
+
+        :param other1: first profile
+        :param other2: second profile
+        :type other1: BaseColumnProfiler
+        :type other2: BaseColumnProfiler
+        """
+        if other1.name == other2.name:
+            self.name = other1.name
+        else:
+            raise ValueError("Text names unmatched: {} != {}"
+                             .format(other1.name, other2.name))
+
+        self.times = defaultdict(
+            float, {key: (other1.times.get(key, 0)
+                          + other2.times.get(key, 0)
+                          + self.times.get(key, 0))
+                    for key in (set(other1.times) | set(other2.times)
+                                | set(self.times))}
+        )
+
+        self.sample_size = other1.sample_size + other2.sample_size
+
+    def __add__(self, other):
+        """
+        Merges the properties of two TextProfiler profiles
+
+        :param self: first profile
+        :param other: second profile
+        :type self: TextProfiler
+        :type other: TextProfiler
+        :return: New TextProfiler merged profile
+        """
+        if not isinstance(other, TextProfiler):
+            raise TypeError("Unsupported operand type(s) for +: "
+                            "'TextProfiler' and '{}'".format(
+                other.__class__.__name__))
+        merged_profile = TextProfiler(None)
+        
+        if self.name == other.name:
+            merged_profile.name = self.name
+        else:
+            raise ValueError("Text names unmatched: {} != {}"
+                             .format(self.name, other.name))
+
+        merged_profile.times = defaultdict(
+            float, {key: (self.times.get(key, 0)
+                          + other.times.get(key, 0))
+                    for key in (set(self.times) | set(other.times))}
+        )
+        
+        BaseColumnProfiler._merge_calculations(merged_profile.__calculations,
+                                 self.__calculations,
+                                 other.__calculations)
+
+        if "vocab" in merged_profile.__calculations:
+            merged_profile.vocab = self.vocab.copy()
+            merged_profile._update_vocab(other.vocab)
+            
+        if "words" in merged_profile.__calculations:
+            merged_profile.word_count = self.word_count.copy()
+            for word in other.word_count:
+                if word in merged_profile.word_count:
+                    merged_profile.word_count[word] += other.word_count[word]
+                elif word not in self._stop_words:
```

Comment:
stop words are constant time, we might want to check it first which might make it faster if self.word_count is very large. Additionally, if word_count is a defaultdict(int) we can just add it no matter if it was there previously or not.
i.e.
```python
for word in other.word_count:
    if word not in self._stop_words:
        merged_profile.word_count[word] += other.word_count[word]
```

Reply:
Fixed